vendor/k8s.io/kubernetes/pkg/kubectl/cmd/drain.go

```diff
@@ -208,27 +222,47 @@ func (o *DrainOptions) SetupDrain(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	o.nodeInfos = []*resource.Info{}
 	o.mapper, o.typer = o.Factory.Object()
 
 	cmdNamespace, _, err := o.Factory.DefaultNamespace()
 	if err != nil {
 		return err
 	}
 
-	r := o.Factory.NewBuilder(true).
+	nameArgs := []string{"nodes"}
+
+	b := o.Factory.NewBuilder(true).
 		NamespaceParam(cmdNamespace).DefaultNamespace().
-		ResourceNames("node", args[0]).
-		Do()
+		SelectorParam(o.Selector)
+
+	if len(args) > 0 {
+		nameArgs = append(nameArgs, args[0])
+	}
 
+	r := b.
+		ResourceTypeOrNameArgs(true, nameArgs...).
+		Flatten().
+		Do()
 	if err = r.Err(); err != nil {
 		return err
 	}
 
+	if len(o.Selector) > 0 {
+		infos, err := r.Infos()
+		if err != nil {
+			return err
+		}
+		if len(infos) == 0 {
+			return fmt.Errorf("error: no nodes matched selector %q", o.Selector)
```

Comment:
Do you think we need that message? I haven't seen this being error-ed out anywhere else when applying label selector.

Reply:
Thanks, will remove